modules/standard/IO.chpl

```diff
@@ -2728,6 +2729,13 @@ private proc _read_text_internal(_channel_internal:qio_channel_ptr_t,
     var ret = qio_channel_scan_string(false, _channel_internal, tx, len, -1);
     x = new string(tx, length=len, needToCopy=false);
     return ret;
+  } else if t == _bytes {
+    // handle _bytes
+    var len:int(64);
+    var tx: c_string;
+    var ret = qio_channel_scan_string(false, _channel_internal, tx, len, -1);
```

Comment:
Oh, I see, the qio_channel_scan_string / qio_channel_print_string handle various forms of escaping. Perhaps it is reasonable for these to do that escaping for bytes strings too.

 * if we want writef("%ht", bytesContainingOneZeroByte) to emit an escaped `\x00` or `\u0000` then it should call qio_channel_scan_string / qio_channel_print_string and you should add an argument to indicate if the string is UTF-8 or bytes. If it's bytes, you should read adjust the method to not call read_char / write_char (these would do character-set conversion) and rather call qio_channel_read_byte / qio_channel_write_byte (or just read/write with a 1 byte buffer).
 * if we want writef("%ht", bytesContainingOneZeroByte) to emit just the null byte, then it should call qio_channel_read_string / qio_channel_write_string, and we should do our best in the formatted I/O code to make sure that the various format details (padded width, etc) apply reasonably well to the bytes.

Reply:
I think direction-wise, I want the first option. As, in general, it'd be helpful to printout `bytes` without creating a huge mess because of non-conforming bytes. In such cases we should instead use `\x00` or similar. I will see if I can get this right in the implementation.